src/directives/style-map.ts

```diff
@@ -52,15 +52,18 @@ export const styleMap = directive((styleInfo: StyleInfo) => (part: Part) => {
   const {committer} = part;
   const {style} = committer.element as HTMLElement;
 
-  // Handle static styles the first time we see a Part
-  if (!styleMapCache.has(part)) {
+  let previousStyleProperties = previousStylePropertyCache.get(part);
+
+  if (previousStyleProperties === undefined) {
+    // Write static styles once
     style.cssText = committer.strings.join(' ');
+    previousStylePropertyCache.set(part, previousStyleProperties = new Set());
   }
 
   // Remove old properties that no longer exist in styleInfo
-  const oldInfo = styleMapCache.get(part);
-  for (const name in oldInfo) {
+  for (const name of previousStyleProperties) {
     if (!(name in styleInfo)) {
+      previousStyleProperties.delete(name);
```

Comment:
I wonder if this will cause deopts in the iteration.

Reply:
`previousStyleProperties` is a Set, so no. Sets and Maps don't have shapes and are effectively always in "dictionary" mode.